File: backend/api/utils/telefono.util.js
const { sanitizarInput } = require('./helpers');

/**
 * Normaliza teléfono argentino: solo dígitos.
 */
function soloDigitos(telefono) {
  return String(telefono || '').replace(/\D/g, '');
}

/**
 * Valida teléfono para registro de comercio (obligatorio, mín. 8 dígitos).
 */
function validarTelefonoComercio(telefono) {
  const raw = sanitizarInput(telefono, 50);
  if (!raw) return { ok: false, mensaje: 'El teléfono es obligatorio para comercios' };
  const digits = soloDigitos(raw);
  if (digits.length < 8) {
    return { ok: false, mensaje: 'El teléfono debe tener al menos 8 dígitos (ej: 3434567890)' };
  }
  if (digits.length > 15) {
    return { ok: false, mensaje: 'El teléfono es demasiado largo' };
  }
  return { ok: true, normalizado: digits };
}

module.exports = { soloDigitos, validarTelefonoComercio };

File: app/src/main/java/com/piku/app/utils/TelefonoUtil.kt
package com.piku.app.utils

object TelefonoUtil {

    fun soloDigitos(texto: String): String = texto.filter { it.isDigit() }

    /**
     * Valida teléfono argentino para comercio (obligatorio, mín. 8 dígitos).
     */
    fun validarComercio(telefono: String): String? {
        val digits = soloDigitos(telefono.trim())
        if (digits.isEmpty()) return "El teléfono es obligatorio para comercios"
        if (digits.length < 8) return "El teléfono debe tener al menos 8 dígitos (ej: 3434567890)"
        if (digits.length > 15) return "El teléfono es demasiado largo"
        return null
    }
}
